File: zenith/echo-market-bloom-main/echo-market-bloom-main/src/main/java/com/ecommerce/service/WishlistService.java

package com.ecommerce.service;

import com.ecommerce.dto.WishlistItemDto;

import java.util.List;

public interface WishlistService {
    List<WishlistItemDto> getWishlistItems();
    WishlistItemDto addToWishlist(Long productId);
    void removeFromWishlist(Long productId);
    boolean isProductInWishlist(Long productId);
}

File: zenith/echo-market-bloom-main/echo-market-bloom-main/src/services/wishlistService.js

import ApiService from './apiService';
import { API_ENDPOINTS } from '../config/apiConfig';

class WishlistService {
  static getWishlistItems() {
    return ApiService.get(API_ENDPOINTS.WISHLIST.ITEMS);
  }
  
  static addToWishlist(productId) {
    return ApiService.post(API_ENDPOINTS.WISHLIST.ADD, { productId });
  }
  
  static removeFromWishlist(productId) {
    return ApiService.delete(`${API_ENDPOINTS.WISHLIST.REMOVE}/${productId}`);
  }
  
  static isProductInWishlist(productId) {
    return ApiService.get(`${API_ENDPOINTS.WISHLIST.ITEMS}/${productId}/exists`)
      .then(response => response.data)
      .catch(() => false);
  }
}

export default WishlistService;
